### ISSUE TITLE
[Enhancement]: Upgrade Testcontainers.Pulsar to use Apache Pulsar v4.0.0

### ISSUE THREAD

**Issue by keshramjith** [testcontainers/testcontainers-dotnet#1291]
### Problem

Upgrade docker image of Testcontainers.Pulsar to Apache Pulsar v4.0.0

### Solution

Changing docker image version and any other necessary changes required for Apache Pulsar v4.0.0

### Benefit

Using the latest [LTS](https://pulsar.apache.org/blog/2024/10/24/announcing-apache-pulsar-4-0/) version of Apache Pulsar

### Alternatives

None

### Would you like to help contributing this enhancement?

Yes

**Comment by keshramjith** [comment]
Please add the module label, I'm not sure how to do that

**Comment by HofmeisterAn** [comment]
Thanks for creating the issue. Do you know if any changes to the module are necessary? Could you simply override the default image instead? Typically, we try to avoid changing the default image version to prevent introducing breaking changes; instead, we recommend developers to pin the version.

**Comment by keshramjith** [comment]
Yes, I did try overriding the image with `.WithImage()`, however I encountered a bunch of issues and some module changes are required.

**Comment by HofmeisterAn** [comment]
I see, thanks for the clarification. In this case, we usually make the necessary changes but try to support both configurations and ensure backward compatibility.
